yarn/src/main/scala/org/apache/spark/deploy/yarn/YarnAllocator.scala

```diff
@@ -169,6 +172,24 @@ private[yarn] class YarnAllocator(
   }
 
   /**
+   * Init `executorIdCounter`
+   */
+  def initExecutorIdCounter(): Unit = {
+    val port = sparkConf.getInt("spark.yarn.am.port", 0)
+    SparkHadoopUtil.get.runAsSparkUser { () =>
+      val init = RpcEnv.create(
+        "executorIdCounterInit",
+        Utils.localHostName,
+        port,
+        sparkConf,
+        new SecurityManager(sparkConf))
+      val driver = init.setupEndpointRefByURI(driverUrl)
```

Comment:
Hi @jerryshao , thanks for your comments. I see what you mean, I will fix it soon. Thanks a lot.